dashboard/cron.py

```diff
@@ -417,7 +417,17 @@ def update_resource_access(self):
             logger.debug(f'resource_access_df:\n'
                          f'{resource_access_df}\n'
                          f'{resource_access_df.dtypes}')
-
+            # only keep access events generated by students
+            student_enrollment_type= User.EnrollmentType.STUDENT
+            student_enrollment_df = pd.read_sql(
+                'select user_id, course_id from user where enrollment_type= %s',
+                engine, params={student_enrollment_type})
+            resource_access_df = pd.merge(
+                resource_access_df, student_enrollment_df,
+                on=['user_id', 'course_id'],
+                # use inner merge to keep only resource access event (left)
+                # innitiated by people with student enrollment type (right)
+                how='inner')
```

Comment:
Because I was a bozo and forgot to update my queries, I found an edge case that I already shared with @zqian where if the query results initially fed into`resource_access_df` only contain `-1` (and thus are converted via the earlier merge to string user IDs), then a `pandas` error is thrown like this:

```
You are trying to merge on object and int64 columns. If you wish to proceed you should use pd.concat
```

This seems to be because the two `user_id` fields in the merge here have types of `object` and `int64`. The easiest way to reproduce this is to only query for MiVideo resource access events (ensuring that some data will be returned for courses). I however didn't see this in the mixed case where MiVideo, Canvas, and leccap queries are in use; maybe `pandas` is smart enough to convert to `int64` if it seems some of them fit that description.

It may be worthwhile to add a line like this around 374 to convert the user ID column back to `int64` so this merge can handle this edge case -- so long as it doesn't create any other problems 🤣 

```
resource_access_df['user_id'] = pd.to_numeric(resource_access_df['user_id'])
```

Reply:
Looks like we found that this mismatch of types actually does cause a substantial problem whenever MiVideo data is used (alone or in conjunction with other queries). The inner join seems to remove all MiVideo data, probably because the data type of the IDs is still a string. The above line seems to fix this.

Reply:
added the above line change.